packages/superset-ui-connection/types/SupersetClient/index.d.ts

```diff
@@ -0,0 +1,76 @@
+type body = RequestInit['body'];
+type cache = RequestInit['cache'];
+type credentials = RequestInit['credentials'] | void;
+type csrfToken = string;
+type endpoint = string;
+type headers = { [key: string]: string };
+type host = string;
+type method = RequestInit['method'];
+type protocol = 'http:' | 'https:' | 'file:';
+type postPayload = { [key: string]: any };
+type mode = RequestInit['mode'];
+type redirect = RequestInit['redirect'];
+type timeout = number | void;
+type parseMethod = 'json' | 'text';
+type signal = RequestInit['signal'];
+type stringify = boolean;
+type url = string;
+
+interface ClientConfig {
+  credentials?: credentials;
+  csrfToken?: csrfToken;
+  headers?: headers;
+  host?: host;
+  protocol?: protocol;
+  mode?: mode;
+  timeout?: timeout;
+}
+
+interface CallApi {
```

Comment:
aside from the `type`s which are re-used across `interface`s, some of these seem slightly more useful within the files themselves. thoughts?